### PR TITLE
Now on process termination the composer saves it's state. Also some mailbox optimizations.

### PR DESCRIPTION
**Pull request by jorgeblacio**

### CODE REVIEW

File: src/main/kotlin/com/criptext/mail/scenes/composer/ComposerController.kt [hunk 1]
```diff
@@ -610,7 +611,7 @@ class ComposerController(private val storage: KeyValueStorage,
         if(model.isReplyOrDraft || model.isSupport){
             scene.setFocusToComposer()
         } else {
-            scene.setFocusToTo()
+            if(model.to.isEmpty()) scene.setFocusToTo() else scene.setFocusToComposer()
```
Comment (danieltigse) [inline src/main/kotlin/com/criptext/mail/scenes/composer/ComposerController.kt]: In what scenario the `to` is empty and is not a reply or draft or support?

File: src/main/kotlin/com/criptext/mail/scenes/mailbox/workers/SendMailWorker.kt [hunk 2]
```diff
@@ -339,7 +337,8 @@ class SendMailWorker(private val signalClient: SignalClient,
         return when (finalResult) {
             is Result.Success -> {
                 db.increaseContactScore(listOf(emailId))
-                MailboxResult.SendMail.Success(emailId, isSecure)
+                val thread = db.getEmailThreadFromEmail(currentEmail!!, Label.LABEL_INBOX, Label.defaultItems.rejectedLabelsByFolder(Label.LABEL_INBOX).map { it.id }, activeAccount.userEmail, activeAccount)
```
Comment (danieltigse) [inline src/main/kotlin/com/criptext/mail/scenes/mailbox/workers/SendMailWorker.kt]: what happens If I reply to an email that is not from INBOX?

File: src/main/kotlin/com/criptext/mail/BaseActivity.kt [hunk 3]
```diff
@@ -148,9 +150,19 @@ abstract class BaseActivity: PinCompatActivity(), IHostActivity {
                     )
                 }
                 COMPOSER_MODEL -> {
-                    ComposerModel(
+                    val composerModel = ComposerModel(
                             type = ComposerType.fromJSON(savedInstanceState.getString("composerType")!!, this)
                     )
+                    composerModel.subject = savedInstanceState.getString("subject")!!
```
Comment (danieltigse) [inline src/main/kotlin/com/criptext/mail/BaseActivity.kt]: I am not sure about `!!`

File: src/main/kotlin/com/criptext/mail/scenes/composer/ComposerController.kt (hunk 1 above)
Comment (jorgeblacio) [reply src/main/kotlin/com/criptext/mail/scenes/composer/ComposerController.kt]: When the app recovers from a system termination.

File: src/main/kotlin/com/criptext/mail/BaseActivity.kt (hunk 3 above)
Comment (jorgeblacio) [reply src/main/kotlin/com/criptext/mail/BaseActivity.kt]: It's validated at the beginning of the when statement.

**Pull request merged**